pytest_celery/message_brokers/redis_broker.py

```diff
@@ -1,23 +1,23 @@
+from __future__ import annotations
+
 from functools import cached_property
 from typing import List
 
 from kombu import Queue
+from redis.client import Redis
 from testcontainers.redis import RedisContainer
 
 from pytest_celery.message_brokers.message_broker import MessageBroker
 
 
 class RedisBroker(MessageBroker):
     @property
-    def queues(self) -> List[Queue]:
+    def queues(self) -> list[Queue]:
```

Comment:
This only works on Python 3.9+. See [the depreciation notice at the bottom of the section](https://docs.python.org/3/library/typing.html#typing.List)

Instead, you can create a compat.py module In it, you check if `list[Queue]` raises an exception. If it does, set `List` to `list`, if it doesn't set `List` to `typing.List` and import the type from that module.

This is how a lot of compatibility code is written.

```python
try:
  list[object]
except AttributeError:
  List = typing.List
else:
  List = list
```

I think that the code above raises an `AttributeError`. You need to verify first.

Reply:
@thedrow Are you saying that there is no single generic list object in Python that can be used in all versions?